src/arch/ia32e/address.rs

```diff
@@ -0,0 +1,84 @@
+pub struct VirtualAddress(u64);
+
+impl VirtualAddress {
+    /// Try to convert u64 into an ia32e virtual address.
+    ///
+    /// # Safety
+    ///
+    /// This function ensures the virtual address is correct.
+    pub fn try_new(addr: u64) -> Option<Self>
```

Comment:
A result would fit better this constructor.

Reply:
Done